components/org.wso2.carbon.identity.tenant.resource.manager/src/main/java/org/wso2/carbon/identity/tenant/resource/manager/TenantAwareAxis2ConfigurationContextObserver.java

```diff
@@ -0,0 +1,238 @@
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.getResourceManager;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.handleServerException;
+import static org.wso2.carbon.identity.tenant.resource.manager.util.ResourceUtils.populateMessageWithData;
+
+/**
+ * Axis2Observer for generating tenant wise publisher configurations.
+ */
+public class TenantAwareAxis2ConfigurationContextObserver extends AbstractAxis2ConfigurationContextObserver {
+
+    private static final Log log = LogFactory.getLog(TenantAwareAxis2ConfigurationContextObserver.class);
+
+    /**
+     * Add the tenant wise publisher and stream Configuration in tenant loading.
+     *
+     * @param tenantId tenant ID.
+     */
+    @Override
+    public void creatingConfigurationContext(int tenantId) {
+
+        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+        log.info("Loading configuration context for tenant domain: " + tenantDomain);
+        loadEventStreamAndPublisherConfigurations(tenantId, tenantDomain);
+    }
+
+    /**
+     * This method loads event stream and publisher configurations for a specific tenant.
+     *
+     * @param tenantId     tenant id.
+     * @param tenantDomain tenant domain.
+     */
+    private void loadEventStreamAndPublisherConfigurations(int tenantId, String tenantDomain) {
+
+        List<EventPublisherConfiguration> activeEventPublisherConfigurations;
+        List<EventStreamConfiguration> eventStreamConfigurationList;
+        try {
+            startSuperTenantFlow();
+            activeEventPublisherConfigurations = getSuperTenantEventPublisherConfigurations();
+            eventStreamConfigurationList = getSuperTenantEventStreamConfigurations();
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+        try {
+            startTenantFlow(tenantId, tenantDomain);
+            createTenantEventStreamConfiguration(eventStreamConfigurationList);
+            createTenantPublisherConfigurationFromConfigStore();
+
+            if (activeEventPublisherConfigurations != null) {
+                createTenantPublisherConfigurationFromSuperTenantConfig(activeEventPublisherConfigurations);
+            }
+        } finally {
+            PrivilegedCarbonContext.endTenantFlow();
+        }
+    }
+
+    /**
+     * This method creates publisher configurations tenant wise by fetching them from configuration store.
+     */
+    private void createTenantPublisherConfigurationFromConfigStore() {
+
+        List<ResourceFile> tenantSpecificPublisherFiles;
+        try {
+            tenantSpecificPublisherFiles = TenantResourceManagerDataHolder.getInstance().getConfigurationManager()
+                    .getFiles(PUBLISHER);
+            for (ResourceFile resourceFile : tenantSpecificPublisherFiles) {
```

Comment:
Can 'tenantSpecificPublisherFiles' be null ?

Reply:
fixed with collectionUtils